### PR TITLE
#585 dynamic sky toggle

### LINKED ISSUE #585: Dynamic Sky Toggle
**How will this enhancement improve the game?**
While testing the time and lighting system, I noticed that putting a static skybox instead of a procedural one makes quite a noticeable difference in fps. I propose adding another toggle to the time and lighting settings that will toggle whether a dynamic sky is enabled or not. When not enabled, the skybox will be one of 3, either a dawn, afternoon or night skybox, that will be applied at certain times in the day.

**Describe the solution you'd like**

- [x] Adding a "Dynamic Sky Toggle" that enables/disables the dynamic sky.

**Additional context** (Remove if not relevant)
This toggle will help with performance of the new dynamic lighting implemented in AoTTG2. I have already done a small test run of this in the #113 branch, it would not be a lot of effort to implement it.

**Contributor instructions** (Do not modify)
1. Determine if the enhancement is compatible with existing features
2. If the idea is denied, please provide a reason and close the issue. If the idea is accepted, consult your team to do an estimation of effort. Remove the [PROPOSAL] name and label.
3. Move the issue to the Future Milestone. On rare exceptions an issue may be moved to Release if there's sufficient upvotes.
4. Once the issue matches the current milestone, follow development procedures.

### PR DESCRIPTION
**Pull request by FAStre**
A new graphics setting "Static Sky Toggle"  has been added to graphics settings that replaces the procedural skybox with a static or HDR one. The toggle significantly improves performance and makes builds with the new lighting system more playable.

The static skybox graphics setting is persisted via playerprefs and the DayAndNightController prefab sets the skybox to an HDR(static) one depending on the CurrentTime variable. closes #585

### CODE REVIEW

File: Assets/Scripts/DayNightCycle/ToggleStaticSkybox.cs
Comment (Lithrun) [inline Assets/Scripts/DayNightCycle/ToggleStaticSkybox.cs:7]: Don't forget about the access modifiers. These must always be explicitly defined to improve readability

File: Assets/Scripts/DayNightCycle/DayAndNightControl.cs
Comment (Lithrun) [inline Assets/Scripts/DayNightCycle/DayAndNightControl.cs:167]: Keep the property at the left side. Switching it between sides can be confusing.

If 5 is less than CurrentTime AND CurrentTime is less than or equal to 8, then...

More readable:
If CurrentTime is Greater than 5 AND Current Time is less than or equal to 8

File: Assets/Scripts/DayNightCycle/DayAndNightControl.cs
Comment (Lithrun) [inline Assets/Scripts/DayNightCycle/DayAndNightControl.cs]: Magic strings should be in a constant

**Review by Lithrun — CHANGES_REQUESTED** [review]
Don't forget your access modifiers. They help a lot

**Review by Lithrun — APPROVED** [review]

**Comment by Lithrun** [comment]
Issue is approved, but will wait until #537 is merged.

**Comment by FAStre** [comment]
> Issue is approved, but will wait until #537 is merged.

537 has been merged here and everything is working as on development branch

**Comment by Lithrun** [comment]
> 
> 
> > Issue is approved, but will wait until #537 is merged.
> 
> 537 has been merged here and everything is working as on development branch

Alright, will clone the branch, perform some sanity checks, and then see if something explodes.

**Pull request merged**
